py/desitarget/mock/build.py

```diff
@@ -1042,3 +1060,509 @@ def join_targets_truth(mockdir, outdir=None, force=False, comm=None):
+    if target_name == 'sky':
+        select_targets_function(targets, truth)
+        return [targets, truth]
+
+    truth['TRUEZ'][:] = source_data['Z'][onbrick]
+
+    # Assign the magnitudes
+    meta = getattr(Magnitudes, target_name)(source_data, index=onbrick, mockformat=mockformat)
+
+    for key in ('TEMPLATEID', 'MAG', 'FLUX_G', 'FLUX_R', 'FLUX_Z', 'FLUX_W1', 'FLUX_W2', 
+                'OIIFLUX', 'HBETAFLUX', 'TEFF', 'LOGG', 'FEH'):
+        truth[key][:] = meta[key]
+
+    # Perturb the photometry based on the variance on this brick and apply
+    # target selection.
+    for band, fluxkey in enumerate( ('FLUX_G', 'FLUX_R', 'FLUX_Z', 'FLUX_W1', 'FLUX_W2') ):
+        targets[fluxkey][:] = truth[fluxkey][:] + rand.normal(scale=onesigma[band], size=nobj)
+
+    if False:
+        import matplotlib.pyplot as plt
+        def elg_colorbox(ax):
+            """Draw the ELG selection box."""
+            from matplotlib.patches import Polygon
+            grlim = ax.get_ylim()
+            coeff0, coeff1 = (1.15, -0.15), (-1.2, 1.6)
+            rzmin, rzpivot = 0.3, (coeff1[1] - coeff0[1]) / (coeff0[0] - coeff1[0])
+            verts = [(rzmin, grlim[0]),
+                     (rzmin, np.polyval(coeff0, rzmin)),
+                     (rzpivot, np.polyval(coeff1, rzpivot)),
+                     ((grlim[0] - 0.1 - coeff1[1]) / coeff1[0], grlim[0] - 0.1)
+                     ]
+            ax.add_patch(Polygon(verts, fill=False, ls='--', color='k'))
+        gr1 = -2.5 * np.log10( truth['FLUX_G'] / truth['FLUX_R'] )
+        rz1 = -2.5 * np.log10( truth['FLUX_R'] / truth['FLUX_Z'] )
+        gr = -2.5 * np.log10( targets['FLUX_G'] / targets['FLUX_R'] )
+        rz = -2.5 * np.log10( targets['FLUX_R'] / targets['FLUX_Z'] )
+        
+        fig, ax = plt.subplots()
+        ax.scatter(rz1, gr1, color='red', label='Noiseless Photometry')
+        ax.scatter(rz, gr, alpha=0.5, color='green', label='Noisy Photometry')
+        ax.set_xlim(-0.5, 2)
+        ax.set_ylim(-0.5, 2)
+        elg_colorbox(ax)
+        ax.legend(loc='upper left')
+        plt.show()
+        import pdb ; pdb.set_trace()
+        
+    if 'BOSS_STD' in source_data.keys():
+        boss_std = source_data['BOSS_STD'][onbrick]
+    else:
+        boss_std = None
+    select_targets_function(targets, truth, boss_std=boss_std)
+
+    keep = np.where(targets['DESI_TARGET'] != 0)[0]
+    nobj = len(keep)
+
+    if nobj == 0:
+        log.warning('No {} targets identified!'.format(target_name.upper()))
+    else:
+        log.debug('Selected {} {}s on brick {}.'.format(nobj, target_name.upper(), thisbrick))
+        targets = targets[keep]
+        truth = truth[keep]
+
+    return [targets, truth]
+
+def targets_truth_no_spectra(params, output_dir='.', realtargets=None, seed=None, verbose=False,
+                  bricksize=0.25, nproc=1, nside=16, healpixels=None):
+    """Generate a catalog of targets, spectra, and the corresponding "truth" catalog
```

Comment:
I would like to keep this as an independent function to use `--no_spectra` regardless of modifications in `targets_truth`.